src/app/scenes/Main/Main.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from "react";
+import NationalInsuranceComparer from "../../components/NationalInsuranceComparer";
+import { Input } from "semantic-ui-react";
+import styled from "@emotion/styled";
+import { useDispatch } from "react-redux";
+import { changeIncome } from "../../store/actions/nationalInsurance";
+import _ from 'lodash'
+
+const MainContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100%;
+  flex-direction: column;
+  min-width: 400px;
+`;
+
+const InsuranceContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const Container = styled.div`
+  min-width: 400px;
+  margin: 30px;
+`;
+
+const InputContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  width: 100%;
+  flex-direction: column;
+`;
+
+const StyledLabel = styled.label`
+  font-weight: bold;
+`
+
+const Main = () => {
+  const dispatch = useDispatch();
+
+  const onChange = useCallback((value) => {
+    dispatch(changeIncome(Number(value) ?? 0))
+  }, [dispatch, changeIncome]) 
+
+  const debouncedChange = _.debounce(onChange, 300)
+
+  return (
+    <MainContainer>
```

Comment:
We liked your UI, clean and functional, particularly like the debounced integration with redux to get a result without the need for extra clicks